### ISSUE TITLE
This appears to be a cross-site request forgery; canceling save.

### ISSUE THREAD

**Issue by oupala** [SemanticMediaWiki/SemanticMediaWiki#1950]
### Setup and configuration

- SMW version: 2.3.1 (and Semantic Forms 3.6)
- MW version: 1.27.1
- DB (MySQL etc.): MySQL

### Issue

Produces a [stack trace](https://www.semantic-mediawiki.org/wiki/Help:Identifying_bugs) and outputs:

```
Définir une propriété
This appears to be a cross-site request forgery; canceling save.
```

Steps to reproduce the observation (recommendation is to use the [sandbox](http://sandbox.semantic-mediawiki.org)):

- install MediaWiki
- install SemanticMediaWiki
- try to create a new property
    - validate the form
    - see the error label: `This appears to be a cross-site request forgery; canceling save.`

I can do any required test.

**Comment by mwjames** [comment]
Thanks.

> SMW version: 2.3.1 (and Semantic Forms 3.6)

I believe this to be a Semantic Forms issue hence it is unrelated to SMW-core. [0] may contain further information.

[0] https://www.mediawiki.org/wiki/Thread:Project:Support_desk/This_appears_to_be_a_cross-site_request_forgery;_canceling_save.

**Comment by mwjames** [comment]
@kghbln FYI

**Comment by oupala** [comment]
Thanks @mwjames for this fast answer. I'll look into Semantic Forms.

I tried to follow the link you mentionned, but I get a "no such thread" answer:

> No such thread
> The thread you specified does not exist. 

Any clue?

**Comment by mwjames** [comment]
> Any clue?

I used the MW search as in [0].

[0] https://www.mediawiki.org/w/index.php?title=Special:Search&search=This+appears+to+be+a+cross-site+request+forgery%3B+canceling+save.&fulltext=Search&profile=threads&searchToken=6sjbve2eccy05f64pmzk9jzuu

**Comment by mwjames** [comment]
>  but I get a "no such thread" answer:

Probably a Flow thing, https://www.mediawiki.org/wiki/Topic:S1ndb9v7dbw9zn0i might work.

**Comment by oupala** [comment]
For information, [this trick](https://www.mediawiki.org/w/index.php?title=Topic:S1ndb9v7dbw9zn0i&topic_showPostId=safhhxiqecwosu4x#flow-post-safhhxiqecwosu4x) seems to work.

> The bypass I found is not "by the book", but it works. Just create the namespaces somewhere else, I put them in the LocalSettings.php by adding these lines:
>
> $wgExtraNamespaces[102] = "Property";
> $wgExtraNamespaces[103] = "Property_talk";
> $wgExtraNamespaces[104] = "Type";
> $wgExtraNamespaces[105] = "Type_talk";
> $wgExtraNamespaces[108] = "Concept";
> $wgExtraNamespaces[109] = "Concept_talk";

Thanks for pointing the right link.

**Comment by mwjames** [comment]
> MW version: 1.27.1
> $wgExtraNamespaces[102] = "Property";

MW 1.27.1 introduced some changes and cost us some grieving hence the recommendation to use 2.4.1 in combination with MW 1.27 (@kghbln Did we note that somewhere?).

[0] https://www.semantic-mediawiki.org/wiki/Semantic_MediaWiki_2.4.1

**Comment by kghbln** [comment]
> MW 1.27.1 introduced some changes and cost us some grieving hence the recommendation to use 2.4.1 in combination with MW 1.27

Issues like https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/1686 already started with MW 1.26 which were fixed with SMW 2.4+

> Did we note that somewhere?

Not so far. I have just [done this.](https://www.semantic-mediawiki.org/w/index.php?title=Semantic_MediaWiki_2.4.1&type=revision&diff=46881&oldid=44624)

**Comment by oupala** [comment]
Thanks for these informations.
